account_banking_mandate/models/res_partner.py

```diff
@@ -21,3 +26,17 @@ def _compute_mandate_count(self):
             for mandate in mandate_data])
         for partner in self:
             partner.mandate_count = mapped_data.get(partner.id, 0)
+
+    @api.multi
+    def compute_valid_mandate_id(self):
+        # Dict to reduce impact with "bug" that process all partners related
+        mandates_dic = {}
+        for partner in self:
+            commercial_partner_id = partner.commercial_partner_id
+            if commercial_partner_id in mandates_dic:
+                partner.valid_mandate_id = mandates_dic[commercial_partner_id]
+            else:
+                mandate_id = partner.commercial_partner_id.bank_ids.mapped(
```

Comment:
This  is incorrect if you have more than one bank account or more than one valid mandate. It should be:
```
mandate_id = partner.commercial_partner_id.mapped('bank_ids.mandate_ids')..filtered(lambda x: x.state == 'valid')[:1].id
```